### PR TITLE
Updated documentation

### PR DESCRIPTION
**Pull request by Dennisbonke**
This PR is a work in progress and aims to update the available documentation. Current focus lies on how to contribute to the project and documenting more of managarm's internal workings.

### CODE REVIEW

File: docs/src/contributing/coding-style.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+- Macros shall begin with an upper case letter, shall be upper case and words shall be separated by an underscore.
+
+## Names
+
+### Use sensible, descriptive names.
+Do not use short cryptic names or names based on internal jokes. It shall be easy to type a name without looking up how it is spelt.
+
+Exception: Loop variables and variables with a small scope (less than 20 lines) may have short names to save space if the purpose of that variable is obvious.
+
+### Only use english names.
```
Comment (ArsenArsen) [inline docs/src/contributing/coding-style.md:24]: it should say English, capital, here, I believe. although I'd ask someone who's better at english (:^)) than me to check the entire document at some point

**Comment by avdgrinten** [comment]
We should get rid of the package list entirely. Just link to xbbs and mention some selected packages (the major ones).

File: docs/src/contributing/overview.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+## Styleguides
+### Coding Style
+
+For the general coding style used in The Managarm Project we refer you to the [coding style](coding-style.md).
+
+
+### Commit Messages
+
+For the general commit message style used in The Managarm Projet we refer you to the [commit message](commit-messages.md)style.
```
Comment (64) [inline docs/src/contributing/overview.md:98]: Projet -> Project, and you're missing a space after the link

File: docs/src/building.md [hunk 3]
```diff
@@ -20,90 +20,123 @@ export MANAGARM_DIR="$HOME/managarm" # In this example we are using $HOME/managa
...
-```sh
-xbstrap install --all
-```
-Note that this command can take **multiple hours**, depending on your machine.
+1.  Initialize the build directory with
+    ```bash
+    cd build
+    xbstrap init ../src
+    ```
+1.  There are several meta-packages available which control what software is build, this means that the build can be started using one of the following commands:
```
Comment (64) [inline docs/src/building.md:68]: We should mention somewhere that svn is required to build

Comment (Dennisbonke) [reply docs/src/building.md:68]: We link to the dockerfile to get a list of dependencies, subversion is listed in there and is thus mentioned

File: docs/src/contributing/overview.md (hunk 2 above)
Comment (Dennisbonke) [reply docs/src/contributing/overview.md:98]: Thanks for spotting that, the joy of writing docs at 2am. I’ll fix it.

File: docs/src/building.md (hunk 3 above)
Comment (64) [reply docs/src/building.md:68]: You need it on the *host* to do the containerized build, and we only say to check the Dockerfile for those doing a non-containerized build, no?

Comment (Dennisbonke) [reply docs/src/building.md:68]: You are correct, my mistake. I'll fix it.

**Comment by avdgrinten** [comment]
This PR has been open for a long time now. Let's merge it and to further improvements in an iterative way.

**Pull request merged**